src/lang/lang_parser.mly

```diff
@@ -293,6 +296,25 @@ expr:
   | BEGIN exprs END                  { $2 }
   | FUN LPAR arglist RPAR YIELDS expr{ mk_fun ~pos:$loc $3 $6 }
   | LCUR exprss RCUR                 { mk_fun ~pos:$loc [] $2 }
+  | expr COALESCE expr               { let null = mk ~pos:$loc($1) (Var "null") in
+                                       let op =  mk ~pos:$loc($1) (Invoke (null, "default")) in
+                                       let handler = mk_fun ~pos:$loc($3) [] $3 in
+                                       mk ~pos:$loc (App (op, ["",$1;"",handler])) }
+  | TRY exprs CATCH bindvar COLON varlist DO exprs END
```

Comment:
If you're planning on adding `in` in the other PR, we should switch to `in` here too instead of `:`. We can do that after both are merged.

Reply:
good idea, in a second time.